### ISSUE TITLE
LGTM.com - false positive `unhashable` alert indexing list with slice

### ISSUE THREAD

**Issue by ntjess** [github/codeql#8711]
Consider the following example (reduced version of the actual alert):
```python
def access_one_or_all(one=False):
    if one:
        accessor = 0
    else:
        accessor = slice(None)
    mylist = [1,2,3]
    return mylist[accessor]
```
This is valid python code, but LGTM gives me the following alert:
`This instance of slice is unhashable.`

**URL to the alert on the project page on LGTM.com**
https://lgtm.com/projects/g/ntjess/s3a/snapshot/8565aa0f07a734d5b92e29b61cd3e6169e67f704/files/s3a/compio/helpers.py#L61

**Comment by aibaars** [comment]
Thanks for your report. It looks like LGTM mistakenly considers `ret` to be a `dict`. When I try the following in the Python interpreter I get:
```
>>> ret = {}
>>> retIdx = slice(None)
>>> ret[retIdx]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'slice'
```
However, for lists using `ret[slice(None)` is indeed perfectly fine.

**Comment by ntjess** [comment]
Thanks for confirming. Is there something I can change in my project to suppress this alert? I.e. would it work to say `ret: list = ...`?

I know I can change project settings etc., but ideally I'd like to get rid of all default alerts in my project

**Comment by aibaars** [comment]
> Thanks for confirming. Is there something I can change in my project to suppress this alert? I.e. would it work to say `ret: list = ...`?

I don't think that will work. It looks like the analysis does not even look at the type. https://github.com/github/codeql/blob/ef0ea247c40da805efa427a37f5213457d18f714/python/ql/src/Expressions/HashedButNoHash.ql#L39

> I know I can change project settings etc., but ideally I'd like to get rid of all default alerts in my project

I guess you could rewrite the code as follows, although that would duplicate the `if`:
```py
def serialize(param: PrjParam, values: t.Sequence[t.Any], returnErrs=True):
  if returnErrs:
    retCnt = 2
  else:
    retCnt = 1
  ret = [pd.Series(name=param, dtype=str).copy() for _ in range(retCnt)]
  if not len(values):
    # Nothing to do
    if returnErrs:
      return ret
    else:
      return ret[0]
```
